core/chainio/avs_writer.go

```diff
@@ -101,14 +111,57 @@ func (w *AvsWriter) SendAggregatedResponse(ctx context.Context, batchMerkleRoot
 		return nil, err
 	}
 
-	receipt, err := utils.WaitForTransactionReceipt(w.Client, ctx, tx.Hash())
+	txNonce := big.NewInt(int64(tx.Nonce()))
+	receipt, err := w.WaitForTransactionReceiptWithIncreasingTip(ctx, tx.Hash(), txNonce, batchMerkleRoot, nonSignerStakesAndSignature)
+	w.logger.Info("Transaction receipt:", "receipt", receipt)
 	if err != nil {
 		return nil, err
 	}
 
 	return receipt, nil
 }
 
+func (w *AvsWriter) WaitForTransactionReceiptWithIncreasingTip(ctx context.Context, txHash gethcommon.Hash, txNonce *big.Int, batchMerkleRoot [32]byte, nonSignerStakesAndSignature servicemanager.IBLSSignatureCheckerNonSignerStakesAndSignature) (*gethtypes.Receipt, error) {
+	for i := 0; i < MaxRetries; i++ {
+		// Attempt to get the transaction receipt
+		receipt, err := utils.WaitForTransactionReceipt(w.Client, ctx, txHash, MaxRetries, SleepTime)
+		if err == nil {
+			return receipt, nil
+		}
+
+		// Simulate the transaction to get the gas limit and gas tip cap again
+		txOpts := *w.Signer.GetTxOpts()
+		txOpts.NoSend = true
+		tx, err := w.AvsContractBindings.ServiceManager.RespondToTask(&txOpts, batchMerkleRoot, nonSignerStakesAndSignature)
+		if err != nil {
+			return nil, err
+		}
+
+		// Use the same nonce as the original transaction
+		txOpts.Nonce = txNonce
+
+		// Increase the gas limit by 10%
+		txOpts.GasLimit = tx.Gas() * 110 / 100
```

Comment:
```suggestion
		Gas limit doesn't need to increase
```

Reply:
Solved in [9db063b](https://github.com/yetanotherco/aligned_layer/pull/330/commits/9db063b1c786d6ebcba13761266ca4848a071d38).